### PR TITLE
Ana/4365 cache api store in db

### LINKED ISSUE #4365: Cache store in db
**Is your feature request related to a problem? Please describe.**

On API startup the api takes a while to connect and therefore it takes a bit to deliver the first data.

**Describe the solution you'd like**

Write the whole store into the db on an update. The db should have one table `storeCaches`. This table should have columns:
- networkId
- store // JSON serialized store

On API start fetch the cached store and populate the store with the cache to reduce waiting times on start.

**Describe alternatives you've considered**

This issue only happens when we deploy the API for a couple of seconds. So maybe we could just ignore the issue.

**Additional context**

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #4365 

**Description:**

Reopening this. Also found some flaws in the new way we handle escaping values in helpers.js. Concretely, what concerns booleans.

EDIT: actually I had to revert the helper to my last commit @faboweb. It wasn't working with your last version.

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: api/lib/database/helpers.js [hunk 1]
```diff
@@ -32,16 +32,18 @@ const graphQLQuery = ({ hasura_url, hasura_admin_key }) => async (query) => {
...
-  if (typeof value === 'object') {
-    const clone = JSON.parse(JSON.stringify(value))
-    Object.keys(clone).forEach((key) => {
-      clone[key] = escapeValue(clone[key])
-    })
-    return JSON.stringify(clone)
-  } else if (typeof value === 'boolean') {
-    return escape(value)
-  } else {
+  try {
```
Comment (faboweb) [inline api/lib/database/helpers.js:35]: let's write a test suite for this.

File: api/lib/database/helpers.js [hunk 2]
```diff
@@ -31,26 +31,25 @@ const graphQLQuery = ({ hasura_url, hasura_admin_key }) => async (query) => {
...
-  } catch (error) {
-    return `"${escape(value)}"`
+    default: return value
   }
 }
 
 function gqlKeyValue([key, value]) {
   // escape all values but handle objects gracefully
-  return `${key}: ${escapeValue(value)}`
+  return `${key}: ${value ? escapeValue(value) : `""`}`
```
Comment (Bitcoinera) [inline api/lib/database/helpers.js:53]: beware the booleans here. I was getting that error, `invalid boolean value, `""``

This was my initial PR description. Maybe I edited it

File: api/tests/source/database.test.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const {escapeValue, gqlKeyValue} = require("../../lib/database/helpers")
+
+describe("database helpers", () => {
+    describe("escape values", () => {
+        it("boolean", () => {
+            expect(escapeValue(true)).toBe(true)
+        })
```
Comment (Bitcoinera) [inline api/tests/source/database.test.js:7]: can you try `false` here? I think it is not going to work

File: api/lib/stores/block-store.js [hunk 4]
```diff
@@ -275,17 +275,8 @@ function enrichValidator(validatorInfo, validator) {
...
-  dbStore.block = store.block
-  dbStore.stakingDenom = store.stakingDenom
-  dbStore.annualProvision = store.annualProvision
-  dbStore.signedBlocksWindow = store.signedBlocksWindow
-  dbStore.validators = store.validators
-  dbStore.proposals = store.proposals
-  dbStore.newValidators = store.newValidators
   await database(config)('').storeStore({
-    store: JSON.stringify(dbStore),
+    store: JSON.parse(JSON.stringify(dbStore)), // clone to remove any functions
```
Comment (Bitcoinera) [inline api/lib/stores/block-store.js]: What functions do you mean?

File: api/tests/source/database.test.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        it("number", () => {
+            expect(escapeValue(42)).toBe(42)
+        })
+        it("object", () => {
+            expect(escapeValue({a: {b: "<script></script>", c: 42}})).toBe(`"{\"a\":{\"b\":\"&lt;script&gt;&lt;/script&gt;\",\"c\":42}}"`)
+        })
+        it("null", () => {
+            expect(gqlKeyValue(["x", undefined])).toBe(`x: ""`)
+        })
```
Comment (Bitcoinera) [inline api/tests/source/database.test.js:22]: it would be better if "null" would be parsed as `null`, no?

File: api/lib/database/helpers.js (hunk 2 above)
Comment (Bitcoinera) [reply api/lib/database/helpers.js:53]: This happens when the Boolean is `false`

Comment (faboweb) [reply api/lib/database/helpers.js:53]: uhhh good catch

File: api/tests/source/database.test.js (hunk 3 above)
Comment (Bitcoinera) [reply api/tests/source/database.test.js:7]: Ah no, ignore. It is working! :tada: bravo!

File: api/lib/database/helpers.js (hunk 2 above)
Comment (Bitcoinera) [reply api/lib/database/helpers.js:53]: ? but the test pass using `false`? :eyes:

Comment (Bitcoinera) [reply api/lib/database/helpers.js:53]: I think it is ok, it goes to default then

File: api/lib/stores/block-store.js (hunk 4 above)
Comment (faboweb) [reply api/lib/stores/block-store.js]: i.e. `addValidatorPictures`

File: api/tests/source/database.test.js (hunk 5 above)
Comment (faboweb) [reply api/tests/source/database.test.js:22]: is that possible?

Comment (Bitcoinera) [reply api/tests/source/database.test.js:22]: should actually go to `default`? ah, but it is actually type string, doesn't convert it...

Comment (faboweb) [reply api/tests/source/database.test.js:22]: not possible to set null

File: api/tests/source/database.test.js (hunk 3 above)
Comment (faboweb) [reply api/tests/source/database.test.js:7]: added a test for that

File: api/tests/source/database.test.js (hunk 5 above)
Comment (Bitcoinera) [reply api/tests/source/database.test.js:22]: but if we don't do JSON.stringify `null` should be `null` anyway, right

**Pull request merged**